knative-operator/pkg/controller/knativeservingobsolete/knativeservingobsolete_controller.go

```diff
@@ -87,31 +93,108 @@ func (r *ReconcileKnativeServingObsolete) Reconcile(request reconcile.Request) (
 			return reconcile.Result{}, err
 		}
 	}
-	// Create the latest CR from the current (previous) CR
-	latest = &servingv1alpha1.KnativeServing{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      current.Name,
-			Namespace: current.Namespace,
-		},
-	}
-	latest.Spec.Config = current.Spec.Config
-	if err := common.Mutate(latest, r.client); err != nil {
+	// Orphan all the children by removing their OwnerRefs
+	if err := r.orphanObsoleteResources(current); err != nil {
 		return reconcile.Result{}, err
 	}
 	// Avoid a certs config conflict in the k-s controller
 	if err := r.removeOldCertsConfig(current.Namespace); err != nil {
 		return reconcile.Result{}, err
 	}
-	// Orphan the kids to avoid webhook race condition
-	if err := r.client.Delete(context.TODO(), current, client.PropagationPolicy(metav1.DeletePropagationOrphan)); err != nil {
+
+	new, err := r.reconcileNewResource(current)
+	if err != nil {
 		return reconcile.Result{}, err
 	}
-	if err := r.client.Create(context.TODO(), latest); err != nil {
-		return reconcile.Result{}, err
+
+	wantStatus := obsolete.KnativeServingStatus{
+		Version:    new.Status.Version,
+		Conditions: deepCopyConditions(new.Status.Conditions),
+	}
+	if !equality.Semantic.DeepEqual(current.Status, wantStatus) {
+		current.Status = wantStatus
+		if err := r.client.Status().Update(context.TODO(), current); err != nil {
+			return reconcile.Result{}, err
+		}
 	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileKnativeServingObsolete) reconcileNewResource(old *obsolete.KnativeServing) (*servingv1alpha1.KnativeServing, error) {
+	new := &servingv1alpha1.KnativeServing{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: old.Namespace, Name: old.Name}, new)
+	if errors.IsNotFound(err) {
+		new := &servingv1alpha1.KnativeServing{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      old.Name,
+				Namespace: old.Namespace,
+			},
+			Spec: servingv1alpha1.KnativeServingSpec{
+				Config: old.Spec.Config,
+			},
+		}
+		if err := common.Mutate(new, r.client); err != nil {
+			return nil, err
+		}
+		if err := r.client.Create(context.TODO(), new); err != nil {
+			return nil, err
+		}
+		return new, nil
+	} else if err != nil {
+		return nil, err
+	} else {
+		want := new.DeepCopy()
+		want.Spec.Config = old.Spec.Config
+		if err := common.Mutate(want, r.client); err != nil {
```

Comment:
Left this here for simplicity. I know it's not cheap but I also didn't really bother to make it smart and make a cheaper check 🤔.